src/test/java/redis/clients/jedis/tests/JedisPoolTest.java

```diff
@@ -42,56 +40,64 @@ public void checkConnections() {
   @Test
   public void checkCloseableConnections() throws Exception {
     JedisPool pool = new JedisPool(new JedisPoolConfig(), hnp.getHost(), hnp.getPort(), 2000);
-    Jedis jedis = pool.getResource();
-    jedis.auth("foobared");
-    jedis.set("foo", "bar");
-    assertEquals("bar", jedis.get("foo"));
-    jedis.close();
+    try (Jedis jedis = pool.getResource()) {
+      jedis.auth("foobared");
+      jedis.set("foo", "bar");
+      assertEquals("bar", jedis.get("foo"));
+    }
     pool.close();
     assertTrue(pool.isClosed());
   }
 
   @Test
   public void checkConnectionWithDefaultPort() {
-    JedisPool pool = new JedisPool(new JedisPoolConfig(), hnp.getHost(), hnp.getPort());
-    Jedis jedis = pool.getResource();
-    jedis.auth("foobared");
-    jedis.set("foo", "bar");
-    assertEquals("bar", jedis.get("foo"));
-    jedis.close();
+    JedisPool pool = new JedisPool(new JedisPoolConfig(), hnp.getHost());
+    try (Jedis jedis = pool.getResource()) {
+      jedis.auth("foobared");
+      jedis.set("foo", "bar");
+      assertEquals("bar", jedis.get("foo"));
+    }
     pool.destroy();
     assertTrue(pool.isClosed());
   }
 
   @Test
-  public void checkJedisIsReusedWhenReturned() {
+  public void checkResourceIsClosableAndReusable() {
+    GenericObjectPoolConfig config = new GenericObjectPoolConfig();
+    config.setMaxTotal(1);
+    config.setBlockWhenExhausted(false);
+    JedisPool pool = new JedisPool(config, hnp.getHost(), hnp.getPort(), 2000, "foobared", 0,
+        "closable-resuable-pool", false, null, null, null);
 
-    JedisPool pool = new JedisPool(new JedisPoolConfig(), hnp.getHost(), hnp.getPort());
     Jedis jedis = pool.getResource();
```

Comment:
Jedis is closeable

Reply:
@gkorland https://github.com/redis/jedis/pull/2325/commits/b9bf1acbab6b9ac9f43813b686d7bacdf5feb097